honeybee_radiance/sensorgrid.py

```diff
@@ -27,9 +30,10 @@ class SensorGrid(object):
         * sensors
         * positions
         * directions
+        * room_identifier
     """
 
-    __slots__ = ('_identifier', '_display_name', '_sensors')
+    __slots__ = ('_identifier', '_display_name', '_sensors', '_room_identifier')
```

Comment:
It should probably be an input for `aperture_group_identifiers`. We can have a method that figures out these groups based on a room_identifier and a model but as I discussed in another comment what really matters is the `aperture_group_identifiers`.

Reply:
I understand the desire to do that but it raises a number of workflow issues, which will ultimately make it very difficult to get this "light path" information in models exported from Grasshopper. Probably the biggest issue it creates is that you will only be able to generate grids that have correct "light path" information after you have a model where you have assigned all apertures and solved all adjacencies. If I generate the grid from a Room before I have added apertures, this "light path" information would basically be empty. Or it could be wrong if I have set apertures but not solved adjacencies. And there's no easy way to figure it out at a later point unless we test that a point is inside a Room volume (which we could do but it seems like this could be unreliable if custom-generated grids are also in the mix).

But, if I store the Room identifier instead of the light path information (like I have done here), I'll be able to figure out the light path correctly upon export, regardless of whether the Room had all of this apertures or adjacencies at the time I created the grid and assigned it to the Room.

I guess if you were ok holding off on the creation/assignment of grids until after you have a complete Model , your proposal could work. But this is probably going to limit the flexibility of working with grids in Grasshopper. So do you agree that the current setup is better at least for now?

Reply:
I guess another option is to support both a `aperture_groups` property and a `room_identifier` property on the SensorGrid. And the hard-specified light path with the `aperture_groups` will override the auto-calculated path from the Room and Model on export. This is a lot like what you suggested in the comment above except that the `aperture_groups` are a property of the grid instead of an option on the info_dict method.

Would you prefer I implement something like that?

Reply:
Yes. That is what I was suggesting except. But in my scenario the `aperture_groups` are not really the full light-path. I think your suggestion to have both is a great starting point and then we will see how it will play out in real cases. I think in most cases that one overwrites the `aperture_groups` they know enough to provide the full light-path.

And while we are here should we call the input and the key in the dictionary `light_path` to be consistent?

Reply:
Just a reminder that whatever we do here should also be done for the view object.

Reply:
I was thinking that `light_path` made more sense as a name, given that it's much more than just `aperture_groups`.

So am I correct in understanding that we should add a `light_path` property to the `SensorGrid` and `View` classes?  And, if this `light_path` is not `None`, the light path is autocalculated using the `room_identifier`. And, if they are both `None`, there's no `light_path` key in the output JSON next to the .pts file so the grid will be run with all aperture groups in the model.

Reply:
I will just assume that my understanding is correct and implement it now.

Reply:
It has been implemented. I'll send a parallel PR over to honeybee-schema.